### ISSUE TITLE
General PDE solving with NNs

### ISSUE THREAD

**Issue by ChrisRackauckas** [SciML/NeuralNetDiffEq.jl#53]
https://arxiv.org/abs/1907.04502

**Comment by KirillZubov** [comment]
Goal list for the prototype for GSoC proposal
1) general ODE solver
2) simple general PDE solver for one dim and first-order task
3) complete and merge second-order derivative PR [#54](https://github.com/JuliaDiffEq/NeuralNetDiffEq.jl/pull/54)
4) example for second-order equation
5) support two-dimensionality

**Comment by ChrisRackauckas** [comment]
The biggest portion of such a project for us would be to make it automatic. The way to do this would be to integrate with the ModelingToolkit.jl PDESystem setup, showcased here: 

https://github.com/JuliaDiffEq/DifferentialEquations.jl/issues/469

Basically it would need to plug into this system and then automatically generate the right neural networks and loss functions from the specification of the PDE.
